src/celestia/celx_internal.h

```diff
@@ -149,8 +291,42 @@ class CelxLua
                         const char* errorMessage = "Boolean argument expected",
                         bool defaultValue = false);
 
-    LuaState* getLuaStateObject();
+    template<typename T> T *safeGetUserData(int index = 0,
+                                            FatalErrors errors = AllErrors,
+                                            const char *errorMessage = "User data expected")
+    {
+        if (!safeIsValid(index))
+            return nullptr;
+        if (!isUserData(index))
+        {
+            if (errors & WrongType)
+                doError(errorMessage);
+            return nullptr;
+        }
+        return reinterpret_cast<T*>(lua_touserdata(m_lua, index));
+    }
+
+    template<typename T> T *safeGetClass(int i,
+                                         FatalErrors fatalErrors = AllErrors,
+                                         const char *msg = "Celx class expected")
+    {
+        T *a = safeGetUserData<T>(i, fatalErrors, msg);
+        if (!isType(i, celxClassId(*a)))
+        {
+            if (fatalErrors & WrongType)
+                doError(msg);
+            return nullptr;
+        }
+        return a;
+    }
 
+    template<typename T> T *getThis(FatalErrors fatalErrors = AllErrors,
```

Comment:
Please replace `getThis` with more meaningful name.

Reply:
Hmm, I have no idea. Legacy <classname>_this was more meaningful?

Reply:
Nope :)

Reply:
But I  don't have any good proposals so if you either don't have any, let's keep getThis.